src/main/java/org/opendatakit/aggregate/submission/type/RepeatSubmissionType.java

```diff
@@ -141,24 +133,33 @@ public void getValueFromEntity(CallingContext cc) throws ODKDatastoreException {
     q.addSort(rel.parentAuri, Direction.ASCENDING); // for GAE work-around
     q.addSort(rel.ordinalNumber, Direction.ASCENDING);
 
-    // reconstruct all the repeating groups from a single submission.
-    // This should be a small number. We don't have the logic to
-    // handle fractional returns of rows.
-    long expectedOrdinal = 1L;
-    List<? extends CommonFieldsBase> repeatGroupList = q.executeQuery();
-    for (CommonFieldsBase cb : repeatGroupList) {
-      DynamicBase d = (DynamicBase) cb;
-      Long ordinal = d.getOrdinalNumber();
-      if ( ordinal == null || ordinal.longValue() != expectedOrdinal ) {
-        String errString = "SELECT * FROM " + d.getTableName()
-            + " WHERE _TOP_LEVEL_AURI = " + d.getTopLevelAuri()
-            + " AND _PARENT_AURI = " + d.getParentAuri() + " is missing a (repeat) group instance OR has an extra copies.";
-        throw new ODKEnumeratedElementException(errString);
-      }
-      ++expectedOrdinal;
-      SubmissionSet set = new SubmissionSet(enclosingSet, d, repeatGroup, form, cc);
-      submissionSets.add(set);
+    long maxOrdinalNumber = 0;
```

Comment:
I'd tend to mark this as `final` but that's a small thing.

Reply:
Would also be good to have a comment above this code block. I think the original one is still relevant?